src/signature.c

```diff
@@ -64,14 +65,36 @@ static char *crl = NULL;
  * returns: true if can initialize and validate certificates, otherwise false */
 bool initialize_signature(void)
 {
+	int ret = -1;
+	time_t mod_sec = 0;
+	struct tm * alttime;
+	struct stat statt;
 	ERR_load_crypto_strings();
 	ERR_load_PKCS7_strings();
 	EVP_add_digest(EVP_sha256());
 
 	if (!get_pubkey()) {
 		goto fail;
 	}
-	if (!validate_certificate()) {
+
+	ret = validate_certificate();
+	if (ret) {
+		if (ret == X509_V_ERR_CERT_NOT_YET_VALID) {
+			/* If we can retrieve an approx. good system time, report out to user */
+			if (stat("/usr/lib/os-release", &statt) != -1) {
+				mod_sec = statt.st_mtim.tv_sec;
+				char timebuf[30];
+				alttime = localtime(&mod_sec);
+				strftime(timebuf, sizeof(timebuf), "%b %d, %Y", alttime);
```

Comment:
I would prefer to use `%F`, which is a more neutral, standardized time/date format. For example, today would be 2016-11-18.